File: 2015/day01.py
test1 = ['(())', 0]
test2 = ['()()', 0]
test3 = ['(((', 3]
test4 = ['(()(()(', 3]
test5 = ['))(((((', 3]
test6 = ['())', -1]
test7 = ['))(', -1]
test8 = [')))', -3]
test9 = [')())())', -3]
test10 = [')', 1]
test11 = ['()())', 5]


def floor(data):
	floor = 0

	for i in range (0, len(data)):
		if data[i] == '(':
			floor += 1
		elif data[i] == ')':
			floor -= 1

	return floor

def basement(data):
	floor = 0

	for i in range (0, len(data)):
		if data[i] == '(':
			floor += 1
		elif data[i] == ')':
			floor -= 1
		if floor == -1:
			return i+1

assert floor(test1[0]) == test1[1]
assert floor(test2[0]) == test2[1]
assert floor(test3[0]) == test3[1]
assert floor(test4[0]) == test4[1]
assert floor(test5[0]) == test5[1]
assert floor(test6[0]) == test6[1]
assert floor(test7[0]) == test7[1]
assert floor(test8[0]) == test8[1]
assert floor(test9[0]) == test9[1]
assert basement(test10[0]) == test10[1]
assert basement(test11[0]) == test11[1]

with open('input1.txt') as file:
	data = file.readline()
	print "Final floor: ", floor(data)
	print "Passing floor -1: ", basement(data)

File: 2017/src/day01.go
package main

import (
	"aoc"
	"fmt"
	"strconv"
	"strings"
)

func find_capcha(input string, part2 bool) int {

	capcha := 0

	chars := strings.Split(input, "")
	step := 1

	if part2 {
		step = len(chars) / 2
	}

	for i := 0; i < len(chars); i++ {
		j := (i + step) % len(chars)

		if chars[i] == chars[j] {
			value, _ := strconv.Atoi(chars[i])
			capcha += value
		}
	}

	return capcha
}

func main() {
	test1 := "1122"
	test2 := "1111"
	test3 := "1234"
	test4 := "91212129"

	aoc.Assert_int(find_capcha(test1, false), 3)
	aoc.Assert_int(find_capcha(test2, false), 4)
	aoc.Assert_int(find_capcha(test3, false), 0)
	aoc.Assert_int(find_capcha(test4, false), 9)

	test5 := "1212"
	test6 := "1221"
	test7 := "123425"
	test8 := "123123"
	test9 := "12131415"

	aoc.Assert_int(find_capcha(test5, true), 6)
	aoc.Assert_int(find_capcha(test6, true), 0)
	aoc.Assert_int(find_capcha(test7, true), 4)
	aoc.Assert_int(find_capcha(test8, true), 12)
	aoc.Assert_int(find_capcha(test9, true), 4)

	input_data := aoc.Read_file("input01.txt")
	fmt.Println(find_capcha(input_data, false))
	fmt.Println(find_capcha(input_data, true))
}
